addons/website_event_sale/controllers/main.py

```diff
@@ -67,6 +68,12 @@ def registration_confirm(self, event, **post):
         # we have at least one registration linked to a ticket -> sale mode activate
         if any(info['event_ticket_id'] for info in registrations):
             order_sudo = request.website.sale_get_order()
+            if order_sudo.partner_id.is_public:
+                contact = tools.formataddr((registrations[0]['name'], registrations[0]['email']))
+                partner = request.env['res.partner'].sudo().find_or_create(contact)
+                if not partner.phone:
+                    partner.phone = registrations[0]['phone']
+                order_sudo.partner_id = partner
```

Comment:
Interesting! If partner exists and gives a different name in the registration, it will just be ignored, right? I believe this is ok though, just asking :)